test/java/org/chromium/net/testing/CronetTestRule.java

```diff
@@ -0,0 +1,359 @@
+    PathUtils.setPrivateDataDirectorySuffix(PRIVATE_DATA_DIRECTORY_SUFFIX);
+    prepareTestStorage();
+    mOldVmPolicy = StrictMode.getVmPolicy();
+    // Only enable StrictMode testing after leaks were fixed in crrev.com/475945
+    if (getMaximumAvailableApiLevel() >= 7) {
+      StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
+                                 .detectLeakedClosableObjects()
+                                 .penaltyLog()
+                                 .penaltyDeath()
+                                 .build());
+    }
+  }
+
+  void tearDown() {
+    if (mCronetTestFramework != null) {
+      mCronetTestFramework.mCronetEngine.shutdown();
+      mCronetTestFramework = null;
+    }
+    try {
+      // Run GC and finalizers a few times to pick up leaked closeables
+      for (int i = 0; i < 10; i++) {
+        System.gc();
+        System.runFinalization();
+      }
+      System.gc();
+      System.runFinalization();
+    } finally {
+      StrictMode.setVmPolicy(mOldVmPolicy);
+    }
+  }
+
+  /**
+   * Starts the CronetTest framework.
+   */
+  public CronetTestFramework startCronetTestFramework() {
+    if (mCronetTestFramework != null) {
+      mCronetTestFramework.mCronetEngine.shutdown();
+    }
+    if (testingJavaImpl()) {
+      ExperimentalCronetEngine.Builder builder = createJavaEngineBuilder();
+      builder.setUserAgent(UserAgent.from(getContext()));
+      mCronetTestFramework = new CronetTestFramework(builder.build());
+      // Make sure that the instantiated engine is JavaCronetEngine.
+      assert mCronetTestFramework.mCronetEngine.getClass() == JavaCronetEngine.class;
+    } else {
+      mCronetTestFramework = new CronetTestFramework(getContext());
+    }
+    return mCronetTestFramework;
+  }
+
+  /**
+   * Creates and returns {@link ExperimentalCronetEngine.Builder} that creates
+   * Java (platform) based {@link CronetEngine.Builder}.
+   *
+   * @return the {@code CronetEngine.Builder} that builds Java-based {@code Cronet engine}.
+   */
+  public ExperimentalCronetEngine.Builder createJavaEngineBuilder() {
+    return (ExperimentalCronetEngine.Builder) new JavaCronetProvider(getContext()).createBuilder();
+  }
+
+  public void assertResponseEquals(UrlResponseInfo expected, UrlResponseInfo actual) {
+    Assert.assertEquals(expected.getAllHeaders(), actual.getAllHeaders());
+    Assert.assertEquals(expected.getAllHeadersAsList(), actual.getAllHeadersAsList());
+    Assert.assertEquals(expected.getHttpStatusCode(), actual.getHttpStatusCode());
+    Assert.assertEquals(expected.getHttpStatusText(), actual.getHttpStatusText());
+    Assert.assertEquals(expected.getUrlChain(), actual.getUrlChain());
+    Assert.assertEquals(expected.getUrl(), actual.getUrl());
+    // Transferred bytes and proxy server are not supported in pure java
+    if (!testingJavaImpl()) {
+      Assert.assertEquals(expected.getReceivedByteCount(), actual.getReceivedByteCount());
```

Comment:
Static import the `assertEquals`?

Reply:
This was like that in the original code - maybe because this is not a test. 

But I agree with you - let's be systematic about import static "assert" methods.

Done.